### ISSUE TITLE
Form component with a header, scrollable body, and sticky footer

### ISSUE THREAD

**Issue by SeanGroff** [SelectQuoteLabs/SQForm#226]
We have a common design pattern that began in our Quote Tools and is now the same design we'll use for Ancillary apps where a whole page is a form.

I've created a POC and ran it by the internal FE team. https://codesandbox.io/s/sqform-full-page-prototype-keisd?file=/src/SQFormFullPage.js

![Screen Shot 2021-05-04 at 9 54 46 AM](https://user-images.githubusercontent.com/4367312/117023480-c4c5e780-acbe-11eb-9371-46eee7c85b6b.png)

The component will likely be named something like SQFormScrollCard, tbd.

A/C

* The form should match the designs
* The form should always take up 100% of it's parent elements available height by making the content area grow/shrink
* Content area should scrollable vertically
* The footer should stick to the bottom, and have 3 components from left to right. Reset Button, InfoText, Submit Button

**Comment by SeanGroff** [comment]
I've been working on this as I'm able to. I know 2 projects are waiting on this component. The Srx project will need this as well. I'll have it released this week.
The component is finished and works within Storybook. 
Next, I need to test this component out on the Data Capture project and the MCE to ensure it works for both use cases. 
Will PR after that.

![Screen Shot 2021-05-17 at 5 01 51 PM](https://user-images.githubusercontent.com/4367312/118562029-95d55a00-b731-11eb-9919-1af45c1ae493.png)

**Comment by SeanGroff** [comment]
Progress Update since I thought I'd be finished by now. 

This component worked perfectly on the Data Capture Project.
Consuming this component on the MCE uncovered some bugs, some of which I still need to figure solutions out for:

Issue 1 - MCE Doesn't specify a static height anywhere. Not a bad thing at all. However, `overflow-y: scroll` will not work unless a static height is specified. Currently, the consumer will have to wrap this component in an element with a static height. 

Issue 2 - The `validationSchema` prop causes an error if we pass it a POJO on the MCE. Currently, we have to pass it a Yup.object().shape() which SQFormScrollable component already does for us.

Issue 3 - When providing a subheader, the content must indent `20px` more.

Further remaining work:

* Write stories in Storybook.
* Write tests (This can be handled in a separate issue that doesn't block this work)

**Comment by SeanGroff** [comment]
I collaborated with @gdoohen1 to bring him up to speed on this feature. He's going to pick up where I left off and get this across the finish line 🏁 
Assigning issue to Grant.

**Comment by gdoohen1** [comment]
Tested this component further on the MCE and addressed the outstanding issues @SeanGroff was seeing:

**Issue 1** - I added two props to the SQFormScrollableCard: `height` and `isSelfBounding`. Height is pretty straightforward, you just provide a `height` and that will set the height for the component (Note: This takes precedence over `isSelfBounding`). When `isSelfBounding` is `true` the component will calculate it's height like so
```
const topOffset = document
    .getElementById(`sqform-scrollable-card-id-${title}`)
    ?.getBoundingClientRect().top;
const calculatedHeight = topOffset
    ? `calc(100vh - ${topOffset}px - 24px)`
    : '100%';
```
What this does is determine the offset (the distance between the top of the component and the top of the page) and subtract that from `100vh`. It also subtracts an additional `24px` for some nice spacing at the bottom. This assumes that there are no components below the `SQFormScrollableCard`, which I think makes sense for our use cases.

The `isSelfBounding` prop seems a bit janky to me, but it was the only way I could find to provide the user some sort of automated height setting by the component itself without requiring an explicit height defined somewhere in the user's DOM tree containing the `SQFormScrollableCard`.

**Issue 2** - This just worked for me when I used a POJO so I can't really take credit for fixing this one =)

**Issue 3** - I added some dynamic padding to the children based on if there is a subheader or not. It looks like
```
childrenContainer: {
      marginLeft: 0,
      marginRight: 0,
      padding: ({hasSubHeader}) => {
        return hasSubHeader ? `${theme.spacing(2)}px ${theme.spacing(4)}px` : 0;
      }
},
```

Here is a screenshot demonstrating how the isSelfBounding prop looks in MCE.
<img width="1723" alt="Screen Shot 2021-05-27 at 1 28 31 PM" src="https://user-images.githubusercontent.com/80288816/119892885-899a8b00-beef-11eb-9075-9e2d45a40400.png">

If you want to test this in MCE pull down https://bitbucket.org/SelectQuote/senior-aqe-ui/branch/scrollable_card_test and delete your `package-lock.json` and your `node_modules` and run `npm i`.
